tensorboard/plugins/interactive_inference/interactive_inference_plugin.py

```diff
@@ -0,0 +1,405 @@
+    """
+    return {
+        '/infer': self._infer,
+        '/update_example': self._update_example,
+        '/examples_from_path': self._examples_from_path_handler,
+        '/sprite': self._serve_sprite,
+        '/duplicate_example': self._duplicate_example,
+        '/delete_example': self._delete_example,
+        '/infer_mutants': self._infer_mutants_handler,
+        '/eligible_features': self._eligible_features_from_example_handler,
+    }
+
+  def is_active(self):
+    """Determines whether this plugin is active.
+
+    Returns:
+      A boolean. Whether this plugin is active.
+    """
+    # TODO(jameswex): Maybe enable if config flags were specified?
+    return False
+
+  def generate_sprite(self, example_strings):
+    # Generate a sprite image for the examples if the examples contain the
+    # standard encoded image feature.
+    self.sprite = (
+        self.create_sprite_image(example_strings)
+        if (len(self.examples) and
+            self.image_feature_name in self.examples[0].features.feature) else
+        None)
+
+  @wrappers.Request.application
+  def _examples_from_path_handler(self, request):
+    """Returns JSON of the specified examples.
+
+    Args:
+      request: A request that should contain 'examples_path' and 'max_examples'.
+
+    Returns:
+      JSON of up to max_examlpes of the tf.train.Examples in the path.
+    """
+    examples_count = int(request.args.get('max_examples'))
+    examples_path = request.args.get('examples_path')
+    try:
+      oss_utils.throw_if_file_access_not_allowed(examples_path,
+                                                 self._logdir,
+                                                 self._has_auth_group)
+      example_strings = oss_utils.example_protos_from_path(
+          examples_path, examples_count, parse_examples=False)
+      self.examples = [
+        tf.train.Example.FromString(ex) for ex in example_strings]
+      self.generate_sprite(example_strings)
+      json_examples = [
+          json_format.MessageToJson(example) for example in self.examples
+      ]
+      self.updated_example_indices = set(range(len(json_examples)))
+      return http_util.Respond(
+          request,
+          {'examples': json_examples,
+           'sprite': True if self.sprite else False}, 'application/json')
+    except common_utils.InvalidUserInputError as e:
+      return http_util.Respond(request, {'error': e.message},
+                               'application/json')
+
+  @wrappers.Request.application
+  def _serve_sprite(self, request):
+    return http_util.Respond(request, self.sprite, 'image/png')
+
+  @wrappers.Request.application
+  def _update_example(self, request):
+    """Updates the specified tf.train.Example.
+
+    Args:
+      request: A request that should contain 'index' and 'example'.
+
+    Returns:
+      An empty response.
+    """
+    if request.method != 'POST':
+      return http_util.Respond(request, {'error': 'invalid non-POST request'},
```

Comment:
You can send statusCode 405 using https://github.com/tensorflow/tensorboard/blob/master/tensorboard/backend/http_util.py#L66

May want to change lines like L242 in this file

Reply:
done